### PR TITLE
refactor: improve custom filters readability with optional chaining

### PR DESCRIPTION
**Pull request by ManikantaMandala**
**Description**
refactor customFilter.js file mentioned here: [these_issues](https://sonarcloud.io/project/issues?issueStatuses=OPEN%2CCONFIRMED&id=asyncapi_generator&open=AZD-TPtRDKL_zKkK1XuB)

also mentioned in this issue #1272

### CODE REVIEW

File: apps/nunjucks-filters/src/customFilters.js [hunk 1]
```diff
@@ -134,52 +134,54 @@ filter.oneLine = oneLine;
...
  * @fieldName {string} - Name of documented property
  * @scopePropName {string} - Name of param for JSDocs
  * @returns {string} JSDoc compatible entry
  */
 function docline(field, fieldName, scopePropName) {
   /* eslint-disable sonarjs/cognitive-complexity */
+  const getType = (f) => f.type() ? f.type() : 'string';
+  const getDescription = (f) => f.description() ? ` - ${f.description().replace(/\r?\n|\r/g, '')}` : '';
+  const getDefault = (f, type) => (f.default() && type === 'string') ? `'${f.default()}'` : f.default();
+  const getPName = (pName) => pName ? `${pName}.` : ``;
```
Comment (derberg) [inline apps/nunjucks-filters/src/customFilters.js:150]:
```suggestion
  const getPName = (pName) => pName ? `${pName}.` : '';
```

**Comment by derberg** [comment]
@ManikantaMandala please followup with failing tests. Next time, you better run them locally to make sure they pass, otherwise you will have to do this test ping pong with maintainers and wait for GitHub Actions to run tests for you

**Comment by ManikantaMandala** [comment]
> @ManikantaMandala please followup with failing tests. Next time, you better run them locally to make sure they pass, otherwise you will have to do this test ping pong with maintainers and wait for GitHub Actions to run tests for you

Thank you for the feedback @derberg 
As this is my first contribution I had made some mistakes
I will do the necessary things for the next pr

**Comment by derberg** [comment]
@ManikantaMandala errors happen even if you make PR number 100, no worries

test run is visible in https://github.com/asyncapi/generator/actions/runs/11219255311/job/31244271269?pr=1275

and yes, run locally as described in development guide - if they pass locally, please share the log in this PR for overview

**Comment by derberg** [comment]
@ManikantaMandala because of a merge of one PR, you have conflicts here now, so please solve them and push updates to the PR

**Review by derberg — COMMENTED** [review]
please remove changes from eslintrc

**Review by derberg — APPROVED** [review]

**Pull request merged**